cmake/modules/AddSwift.cmake

```diff
@@ -198,38 +202,66 @@ function(_add_variant_c_compile_flags)
     RESULT_VAR_NAME result)
 
   is_build_type_optimized("${CFLAGS_BUILD_TYPE}" optimized)
+
+  # MSVC has different options to most compilers!
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    list(APPEND result "-DLLVM_ON_WIN32")
+    list(APPEND result "-D_CRT_SECURE_NO_WARNINGS")
+    # TODO(compnerd) handle /MT
+    list(APPEND result "-D_DLL")
+
+    # Disable C4068: unknown pragma. This means that MSVC doesn't report hundreds of warnings across
+    # the repository for IDE features such as #pragma mark "Title"
+    list(APPEND result "/wd4068")
+  endif()
+
   if(optimized)
     list(APPEND result "-O2")
 
     # Add -momit-leaf-frame-pointer on x86.
-    if("${CFLAGS_ARCH}" STREQUAL "i386" OR "${CFLAGS_ARCH}" STREQUAL "x86_64")
-      list(APPEND result "-momit-leaf-frame-pointer")
+    if("${CFLAGS_ARCH}" STREQUAL "i386" OR "${CFLAGS_ARCH}" STREQUAL "i686" OR "${CFLAGS_ARCH}" STREQUAL "x86_64")
+      if ("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
Yeah, this definitely needs to be a test for MSVC. More compilers are GCC-like than MSVC-like.